.github/workflows/hardhat-core-ci.yml

```diff
@@ -169,6 +169,57 @@ jobs:
           FORCE_COLOR: 3
         run: yarn test:provider
 
+  test-provider-with-edr-rpc-cache:
+    name: Test provider with EDR RPC cache (${{ matrix.os }}, Node ${{ matrix.node }}, ${{ matrix.vm }})
+    runs-on: ${{ matrix.os }}
+    needs: check-rethnet
+    strategy:
+      fail-fast: false
+      matrix:
+        node: [18.15]
+        # disable until actions/virtual-environments#4896 is fixed
+        # os: ["macos-latest", "ubuntu-latest", "windows-latest"]
+        os: ["ubuntu-latest", "windows-latest"]
+        vm: ["rethnet"]
```

Comment:
I'm wondering now whether we could avoid the code duplication by adding a `cache: [true, false]` entry in the matrix, which will consequently be used to determine whether to run the cache step.

Would you be able to try that out?

Reply:
Good it idea, done in [3362561](https://github.com/NomicFoundation/hardhat/pull/4376/commits/33625615c469012c666081c2893f8369dc4c1e21)